Feature/license header

harishp8889 opened the pull request:
# Description
Added new custom action for validating license headers in source code.

# Issues
List the issues impacted by this PR:

| Issue ID |
| -------- |
|   https://github.com/dell/csm/issues/1490       | 

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Tested against fork: https://github.com/harishp8889/test-gofsutil-automation/actions/runs/14713317950/job/41336682755?pr=2

shaynafinocchiaro commented on license-checker/LICENSE-HEADER.txt:
```diff
@@ -0,0 +1,17 @@
+/*
```
Shouldn't it be // instead of
/*
*
*/

harishp8889 replied:
https://github.com/dell/common-github-actions/pull/153#discussion_r2065931453

shaynafinocchiaro commented on .github/workflows/check-license-header.yaml:
```diff
@@ -17,38 +17,34 @@ jobs:
-        run: |
-          curl -L -o .licenserc.yaml https://raw.githubusercontent.com/dell/common-github-actions/refs/heads/feature/license-validation/.github/configs/.licenserc.yaml
+      
+      # - name: Import GPG key
```
Remove this step if its not needed

harishp8889 replied:
Removed

shaynafinocchiaro commented on .github/workflows/check-license-header.yaml:
```diff
@@ -17,38 +17,34 @@ jobs:
-          author_name: License Bot
-          author_email: license_bot@github.com
-          message: "Automatic application of license header"
+      # TODO : we will enable this once autofix is ready
```
Remove this step if its not needed

harishp8889 replied:
Removed

shaynafinocchiaro commented on license-checker/main.go:
```diff
@@ -0,0 +1,136 @@
+/*
```
Shouldn't it be // instead of
/*
*
*/

harishp8889 replied:
@shaynafinocchiaro I see different types of headers are used in different repos below and they are not uniform: 
https://github.com/dell/csi-powermax/blob/main/pkg/file/file.go 
https://github.com/dell/gofsutil/blob/main/gofsutil.go 
The current sample is what I got from @anandrajak1 once we roll out this action. The headers can be fixed

shaynafinocchiaro commented on license-checker/main.go:
```diff
@@ -0,0 +1,136 @@
+		if !hasLicense {
+			fmt.Printf("Missing or incorrect license header: %s\n", file)
+			//  TODO: Uncomment the following line to autofix the license header based on user input
+			// err := autofixLicenseHeader(file, licenseHeader)
```
Should this be uncommented?

harishp8889 replied:
Updated the logic with a flag for auto-fix

shaynafinocchiaro commented on license-checker/main.go:
```diff
@@ -0,0 +1,136 @@
+
+func listGoFiles(root string) ([]string, error) {
+	var files []string
+	err := filepath.Walk(root, func(path string, info os.FileInfo, err error) error {
```
It's more efficient to use filepath.WalkDir https://pkg.go.dev/path/filepath#WalkDir

shaynafinocchiaro commented:
You are committing a binary file under license-checker/license-checker. Can it be removed?

harishp8889 commented:
> You are committing a binary file under license-checker/license-checker. Can it be removed?

Updated the action to use Dockerfile for build and use and removed statically built binary

prablr79 reviewed (approved):
LGTM
